proposals/0000-a11y-cross-platform.md

```diff
@@ -0,0 +1,133 @@
+
+### accessibilityRelationships
+
+This property describes how this component is related to others. Relationship types include:
+
+* labelFor: This component labels the specified component
+* labeledBy: this component is labeled by the specified component
+* describedBy: this component is described by the specified components.
+
+When referring to other components, we may need to add an additional property to view-- namely accessibilityId. We need to find the best forward-looking approach for referencing other components.
+
+#### Android/Fire OS
+
+For labelFor and labeledBy, use the standard Android view and AccessibilityNodeInfo methods.
+
+For describedBy, which is only supported on Fire OS, put the appropriate key/values in the extras bundle.
+
+#### iOS
+
+When associating labels with components, include the label text in the target component's accessibilityLabel.
+
+Don't support describedBy on iOS for now. In future, we could explore adding the accessibilityLabels of the describedBy targets to the accessibilityHint of the component.
+
+### accessibilityRange
+
+This property allows components (such as sliders) to expose range-based information. The value of the property is an object containing the following values:
+
+* min: the minimum value of the range
+* max: the maximum value of the range
+* current: the current value of the component (within the range specified by min and max)
+
+#### Android/Fire OS
+
+This maps directly to Android's RangeInfo interface.
+
+#### iOS
+
+iOS supports a property called accessibilityValue. This value is a simple string. accessibilityRange will be converted to a localized string and placed into accessibilityValue.
+
+## Drawbacks
+
+It will be challenging to ensure that the behavior of these properties is the same between iOS and Android. Building cross-platform properties will likely involve sacrificing a small amount of functionality on each target platform to ensure consistency.
+
+## Alternatives
+
+Components could continue to use the platform-specific properties. This has the drawback of requiring developers to have at least a cursory
+understanding of accessibility on both iOS and Android. And in the case of accessibilityViewIsModal, we expect this to be somewhat difficult to implement correctly on Android.
+
+## Adoption strategy
+
+We don't believe this to be a breaking change. Developers who currently use these properties on Android or iOS can continue to do so, and remove conditional use to allow functionality on other platforms.
+
+## How we teach this
+
+The same documentation that is used to describe the properties on either iOS or Android can be marked as cross-platform for use on either. For the accessibilityRelationships property, labeledBy and labelFor are common paradigms across accessibility support on multiple platforms including web.
+
+## Unresolved questions
```

Comment:
I have a related question (and [issue](https://github.com/necolas/react-native-web/issues/1181)) about content that is hidden visually but visible to screen readers. Typically it will be hidden headings so they show up in the screen reader headings list (e.g., mobile.twitter.com does this) and it is visually hidden with a particular CSS incantation. Would you / how would you do something like this in native apps?

Reply:
Maybe something like this?
```tsx
import React, { useEffect, useState } from 'react';
import { AccessibilityInfo, Text } from 'react-native';

const useIsScreenReaderEnabled = () => {
  const [isScreenReaderEnabled, setIsScreenReaderEnabled] = useState(false);

  useEffect(() => {
    AccessibilityInfo.addEventListener('screenReaderChanged', setIsScreenReaderEnabled);
    AccessibilityInfo.isScreenReaderEnabled().then(setIsScreenReaderEnabled);
    return () => AccessibilityInfo.removeEventListener('screenReaderChanged', setIsScreenReaderEnabled);
  }, []);

  return isScreenReaderEnabled;
};

const Example = () => {
  const isScreenReaderEnabled = useIsScreenReaderEnabled();
  return <Text>I {isScreenReaderEnabled ? 'love' : '❤️'} React Native for Web</Text>;
};
```